src/lms-ds-loader/edfi_lms_ds_loader/loader_facade.py

```diff
@@ -105,9 +105,11 @@ def __get_all_file_paths_callback():
 
 
 def _load_assignments(csv_path: str, db_adapter: MssqlLmsOperations) -> None:
-    sections = set(
-        file_reader.get_all_sections(csv_path)["SourceSystemIdentifier"]
-    )  # we only need to access the last file
+    sections_df = file_reader.get_all_sections(csv_path)
```

Comment:
Be polite instead of crashing if section df is empty, probably because of wrong csv path.  Not that I learned this the hard way...

Reply:
Is there a way to refactor and pass the `sections_df` into these functions, so that we only read once?

Reply:
I'll memoize it. 😎

Reply:
I was creating a shared datastore because I am having the same need. 😆

Reply:
The Python way of memoization: https://dbader.org/blog/python-memoization

TLDR:
`@lru_cache()
def _get_sections_df(csv_path: str) -> pd.DataFrame:
    return file_reader.get_all_sections(csv_path)`

Reply:
Very common in functional programming.

Reply:
Looks very interesting